### ISSUE TITLE
hdfsBuilderConnect does not free the builder

### ISSUE THREAD

**Issue by dtmadko** [PivotalRD/libhdfs3#17]
According to hdfs.h for hdfsFreeBuilder function:

 * It is normally not necessary to call this function since
 * hdfsBuilderConnect frees the builder.

But the connect function does not free the builder.

In JNI based hdfs library, call to hdfsBuilderConnect frees the builder. 

I'd suggest to update the description and leave it up to a user to call hdfsFreeBuilder or call hdfsFreeBuilder in the connect function ( note: this will require changes in hdfsConnectNewInstance and similar functions to prevent "double-free" )

**Comment by wangzw** [comment]
Yes, description of hdfsFreeBuilder() should be changed, users should free the builder when it is not used and more.

This is different from libhdfs in Hadoop. The principle is the one who created the object is responsible to free it after using.

Thanks for your feedback.

**Comment by dtmadko** [comment]
yes, i agree with that. 

I just wanted to point it out because it can cause a memory leak if user does not know about that ( it is how  I discovered it ).

**Comment by kdunn-pivotal** [comment]
Apologies for the *very* delayed response. The development team managing this codebase is also responsible for [Apache HAWQ](http://hawq.incubator.apache.org/) (incubating) and trying to get the first release out the door. As a result, this repo has been unintentionally neglected and recently turned read-only. 

This code was birthed from HAWQ and will be [merged](https://issues.apache.org/jira/browse/HAWQ-1046) back into into that project (now that it is open source as well); [mailing lists](http://mail-archives.apache.org/mod_mbox/incubator-hawq-dev/), pull requests, [issues](https://issues.apache.org/jira/browse/HAWQ) and releases related to libhdfs3 will be there. The plan is to provide [tarball releases](https://issues.apache.org/jira/browse/HAWQ-1058) of this code separate from the larger HAWQ project until it can be relocated into a dedicated repo again (down the road) - see [HAWQ-1059](https://issues.apache.org/jira/browse/HAWQ-1059).

Again, thank you for the contribution and patience while we sort out these logistics. 

@madmatch if this issue/feature is still useful, please respond.
